src/game/gui/hud.cpp

```diff
@@ -184,6 +192,68 @@ void HUD::render() const {
     _debug.render();
 }
 
+void HUD::showCombatHud() {
+    // TODO: make those class members?
+    showControl("BTN_CLEARALL");
+    showControl("BTN_CLEARONE");
+    showControl("BTN_CLEARONE2");
+
+    showControl("LBL_CMBTMODEMSG");
+    showControl("LBL_CMBTMSGBG");
+
+    showControl("LBL_COMBATBG1");
+    showControl("LBL_COMBATBG2");
+    showControl("LBL_COMBATBG3");
+
+    showControl("LBL_QUEUE0");
+    showControl("LBL_QUEUE1");
+    showControl("LBL_QUEUE2");
+    showControl("LBL_QUEUE3");
+}
+
+void HUD::hideCombatHud() {
+    hideControl("BTN_CLEARALL");
+    hideControl("BTN_CLEARONE");
+    showControl("BTN_CLEARONE2");
+
+    hideControl("LBL_CMBTMODEMSG");
+    hideControl("LBL_CMBTMSGBG");
+
+    hideControl("LBL_COMBATBG1");
+    hideControl("LBL_COMBATBG2");
+    hideControl("LBL_COMBATBG3");
+
+    hideControl("LBL_QUEUE0");
+    hideControl("LBL_QUEUE1");
+    hideControl("LBL_QUEUE2");
+    hideControl("LBL_QUEUE3");
+}
+
+void HUD::drawActionQueueItems() const {
+    auto &actionQueue = _game->party().leader()->actionQueue();
+
+    // clear the drawn queue items first
```

Comment:
Maybe add actions first, and clear the rest of the action queue in a loop from `i` until 4?

Reply:
Actually, these two loops are asking to be combined.

```cpp
for (int i = 0; i < 4; ++i) {
    string fill;
    if (it != actionQueue.end()) {
        if ((*it)->type() = ActionType::AttackObject) {
           fill = "i_attack";     
        }
        ++it;
    }
    Control &item = getControl("LBL_QUEUE" + to_string(i));
    item.setBorderFill(fill);
}
```

Reply:
I wanted to account for non-displayable actions in the queue as well (which may be included in the future?) So what about this:
```c++
    auto it = actionQueue.begin();
    for (int i = 0; i < 4; ++i) {
        string fill;

        while (it != actionQueue.end() && (*it)->type() != ActionType::AttackObject)
            ++it;
        if (it != actionQueue.end()) {
            fill = "i_attack";
        }
        Control& item = getControl("LBL_QUEUE" + to_string(i));
        item.setBorderFill(fill);
    }
```